docs/azure/services/Azure_Functions.md

```diff
@@ -0,0 +1,109 @@
+
+There are three Azure Function hosting plans, depending on which different security features are available:
+- Consumption Plan 
+- Premium Plan
+- Dedicated Plan
+- App Service Environment (ASE)
+- Kubernetes (Direct or Azure Arc)
+ASE, Kubernetes, and certain Dedicated Plan options allow further security controls to enforce isolation in the context of compute or networking configuration. This means that users will be able to run function apps in isolated, dedicated, and secure environments.
+
+## Assessment Notes
+### Networking Options
+The available network isolation options will depend on which of the three hosting options offered was used when creating a function. For example the Consumption, Premium and App Service plans run on a multi-tenant infrastructure. Consumption plan has the least number of options for network isolation, whilst App Service and Premium provide more robust controls to restrict network access. 
+
+Hosting a function in an ASE, deploys the function on a single-tenant infrastructure and offers full network isolation such as deploying it in a personal VNet. The network isolation options available for the above mentioned plans are summarised in a table [here] (https://learn.microsoft.com/en-gb/azure/azure-functions/functions-networking-options?tabs=azure-cli).
+
+#### Inbound Access Restrictions
+This option, which is available for all hosting plans, allows users to define a list of IP addresses which are allowed or denied access to a function. When no entries have been defined, an implicit “deny all” is applied.
+
+The following command will output a Function’s inbound access restrictions:
+```CLI
+az functionapp config access-restriction show -g ResourceGroup -n AppName
+```
+#### Outbound Traffic
+##### VNet Integration
+One of the ways to control outbound network access on a function is to set up a Virtual Network integration. This provides a function with the ability to communicate to resources within a dedicated VNet’s subnet. 
+
+This configuration can be found under the `Networking` blade or by running the following command:
+```CLI
+az functionapp vnet-integration list --name --resource-group
+```
+
+Once the VNet integration has been configured, the function will not by default have access to Azure resources. This would require a specific configuration following the change to ensure communication is possible with other internal resources.
+
+There are two VNet integration options available:
+- Regional VNet integration - when the function is connecting to resources in the same region.
+- Gateway-required VNet integration – when the function is connecting to VNets in different regions.
+
+When reviewing Gateway-required VNet integrations, the following security features are important to keep in mind:
+- If a VNet has got peering connections set up with other VNets or VPNs, the connection from the function App will be transitive, causing resources in additional networks to be reachable.
```

Comment:
This is interesting can we share a bit more information regarding this on the transitive bits here? Are we saying that if network A is peered with B and B is peered with C that the function will be able to talk from A -> C? Cause normal peering isn't transitive by nature so this would be interesting if so.